src/eval-frame/actions/language-actions.js

```diff
@@ -94,13 +105,18 @@ export function evaluateLanguagePlugin(pluginText, evalId) {
       sendStatusResponseToEditor("ERROR", evalId);
       return Promise.reject();
     }
-    return loadLanguagePlugin(pluginData, historyId, dispatch).then(() => {
-      sendStatusResponseToEditor("SUCCESS", evalId);
-    });
+    return loadLanguagePlugin(pluginData, historyId, dispatch)
+      .then(() => {
+        sendStatusResponseToEditor("SUCCESS", evalId);
+      })
+      .catch(err => {
+        sendStatusResponseToEditor("ERROR", evalId);
+        return err;
+      });
   };
 }
 
-export function ensureLanguageAvailable(languageId, state, dispatch) {
+export function ensureLanguageAvailable(languageId, state, evalId, dispatch) {
```

Comment:
might be missing it, but it looks like `evalId` is not used in this function?

Reply:
yep that's a mistake, thought I had removed the remaining ones